src/test/ui/issues/issue-69306.stderr

```diff
@@ -89,7 +89,7 @@ error[E0308]: mismatched types
 LL | impl<T> S1<T, u8> {
    |      - this type parameter
 LL |     const C: S1<u8, u8> = Self(0, 1);
-   |                           ^^^^^^^^^^ expected `u8`, found type parameter `T`
+   |                           ^^^^^^^^^^ expected `S1<u8, u8>`, found `S1<T, u8>`
```

Comment:
The label is using the "long types" machinery which will cap the width to at most 15 chars (or revert to the current behavior if that would cause the label to lose meaning). The "hiding matching generics with `_`" logic is not geared towards getting a text output, not can be directly joined with the long types one. I could see if I can make it easy to turn `DiagnosticString` into `String`, and then try them all in order and use whichever is shorter, but I do feel like the current approach "is enough" before bolting more logic onto a single PR (to make it easier to review each step independently).